### ISSUE TITLE
replace show_models() string output with array/list

### ISSUE THREAD

**Issue by LGro** [automl/auto-sklearn#336]
`show_models()` seems to be a quite central method for users to investigate which model has been evaluated to be the most successful. After fitting an ensemble it displays the output of `EnsembleSelection.pprint_ensemble_string()` which is a string representation of an array containing pairs of the weight and the according string representation of the model (see example output below).

What do you think about replacing this output by an array or list data structure (at least at the top level of score & model string)? 
This could enable easier parsing of the results and if printed would still result in a similar output like now.

In case more people like this idea, I'd be happy to implement this and create a pull request.

```
[(0.460000, SimpleClassificationPipeline({'balancing:strategy': 'weighting', 'classifier:__choice__': 'libsvm_svc', 'imputation:strategy': 'mean', 'one_hot_encoding:use_minimum_fraction': 'True', 'preprocessor:__choice__': 'extra_trees_preproc_for_classification', 'rescaling:__choice__': 'standardize', 'classifier:libsvm_svc:C': 2718.215603902964, 'classifier:libsvm_svc:gamma': 0.010295344093308017, 'classifier:libsvm_svc:kernel': 'poly', 'classifier:libsvm_svc:max_iter': -1, 'classifier:libsvm_svc:shrinking': 'True', 'classifier:libsvm_svc:tol': 0.00375915297303114, 'one_hot_encoding:minimum_fraction': 0.0005596685458982962, 'preprocessor:extra_trees_preproc_for_classification:bootstrap': 'False', 'preprocessor:extra_trees_preproc_for_classification:criterion': 'gini', 'preprocessor:extra_trees_preproc_for_classification:max_depth': 'None', 'preprocessor:extra_trees_preproc_for_classification:max_features': 1.0, 'preprocessor:extra_trees_preproc_for_classification:min_samples_leaf': 1, 'preprocessor:extra_trees_preproc_for_classification:min_samples_split': 2, 'preprocessor:extra_trees_preproc_for_classification:min_weight_fraction_leaf': 0.0, 'preprocessor:extra_trees_preproc_for_classification:n_estimators': 100, 'classifier:libsvm_svc:coef0': 0.29331425457977867, 'classifier:libsvm_svc:degree': 4},
dataset_properties={
  'task': 2,
  'sparse': False,
  'multilabel': False,
  'multiclass': True,
  'target_type': 'classification',
  'signed': False})),
(0.440000, SimpleClassificationPipeline({'balancing:strategy': 'none', 'classifier:__choice__': 'libsvm_svc', 'imputation:strategy': 'mean', 'one_hot_encoding:use_minimum_fraction': 'True', 'preprocessor:__choice__': 'no_preprocessing', 'rescaling:__choice__': 'none', 'classifier:libsvm_svc:C': 18469.410087066783, 'classifier:libsvm_svc:gamma': 0.0009052417825455917, 'classifier:libsvm_svc:kernel': 'poly', 'classifier:libsvm_svc:max_iter': -1, 'classifier:libsvm_svc:shrinking': 'False', 'classifier:libsvm_svc:tol': 0.015303710234115636, 'one_hot_encoding:minimum_fraction': 0.010000000000000004, 'classifier:libsvm_svc:coef0': 0.07220421390172338, 'classifier:libsvm_svc:degree': 4},
dataset_properties={
  'task': 2,
  'sparse': False,
  'multilabel': False,
  'multiclass': True,
  'target_type': 'classification',
  'signed': False})),
]
```

**Comment by mfeurer** [comment]
That is a great idea and I'd love to see a pull request on this. Maybe not replacing the current functionality, but adding a second function which returns the ensemble as a list.

**Comment by LGro** [comment]
See pull request #338 for a proposal.

**Comment by mfeurer** [comment]
This feature is available since version 0.2.1. Thanks again for your implementation.
